tbtc-ts/test/sweep.test.ts

```diff
@@ -0,0 +1,460 @@
+import TBTC from "./../src"
+import { BigNumber } from "ethers"
+import { RawTransaction, UnspentTransactionOutput } from "../src/bitcoin"
+import {
+  sweepWithPreviousSweep,
+  sweepWithNoPreviousSweep,
+  testnetDepositScripthashAddress,
+  testnetDepositWitnessScripthashAddress,
+  testnetWalletAddress,
+  testnetWalletPrivateKey,
+} from "./data/bitcoin"
+import { MockBitcoinClient } from "./utils/mock-bitcoin-client"
+// @ts-ignore
+import bcoin from "bcoin"
+import * as chai from "chai"
+import chaiAsPromised from "chai-as-promised"
+chai.use(chaiAsPromised)
+import { expect } from "chai"
+
+describe("Sweep", () => {
+  const fee = BigNumber.from(1600)
+  const firstSweepData = sweepWithNoPreviousSweep
+  const secondSweepData = sweepWithPreviousSweep
+
+  describe("sweepDeposits", () => {
+    let bitcoinClient: MockBitcoinClient
+
+    beforeEach(async () => {
+      bcoin.set("testnet")
+      bitcoinClient = new MockBitcoinClient()
+
+      const rawTransactions = new Map<string, RawTransaction>()
+      for (const utxoData of firstSweepData.utxoData) {
+        rawTransactions.set(utxoData.hash, utxoData.rawTx)
+      }
+      for (const utxoData of secondSweepData.utxoData) {
+        rawTransactions.set(utxoData.hash, utxoData.rawTx)
+      }
+      rawTransactions.set(
+        firstSweepData.sweepResult.transactionHash,
+        firstSweepData.sweepResult.transaction
+      )
```

Comment:
Worth documenting what and why happens here for future-us.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/96/commits/4dccc33c9feb9252e2577f323da7294a691b3569.